File: Programming_6/Validation.py
import re
from urllib.request import urlopen

class Validation:
    @staticmethod
    def decorator_string(func):
        def validate_string(self, value):
            if any(map(str.isdigit, value)):
                raise ValueError("No integers in string!")
            return func(self, value)
        return validate_string

    @staticmethod
    def decorator_number(func):
        def validate_number(self,value):
            try:
                if int(value) < 0:
                    raise ValueError("Number must be > 0")
            except ValueError:
                raise ValueError("No letters/characters in number!")
            return func(self, value)
        return validate_number

    @staticmethod
    def decorator_nomer(func):
        def validate_nomer(self, value):
            try:
                if value[0:4] != '+380' or value[0:4] == '+380' and len(value) > 13 or value[0:4] == '+380' and len(value) < 13:
                    raise ValueError("Phone number must be format +380XXXXXXXXX")
            except ValueError:
                raise ValueError("Phone number must be format +380XXXXXXXXX")
            return func(self, value)
        return validate_nomer

    @staticmethod
    def decorator_url(func):
        def validate_url(self, value):
            try:
                urlopen(value)
            except ValueError:
                raise ValueError("The website_url is not valid")
            return func(self, value)
        return validate_url

    @staticmethod
    def decorator_price(func):
        def validate_price(self, value):
            try:
                x = float(value)
                y = str(x).split('.')
                if len(y[-1]) > 2:
                    raise ValueError("Format of budget is wrong!")
            except ValueError:
                raise ValueError("No letters/characters in number!")
            return func(self, value)
        return validate_price

File: C#/Template collection/Template collection/Validation.cs
using System;
using System.Linq;


namespace Classes
{
    class Validation
    {
        public static string Number(string value)
        {
            int result;
            if (int.TryParse(value, out result))
                return value;
            else
                throw new ArgumentException("No letters/characters in number!");
        }

        public static string String(string value)
        {
            foreach (char c in value)
            {
                if (char.IsDigit(c))
                    throw new ArgumentException("No integers in string!");
            }
            return value;
        }

        public static string Phone(string value)
        {
            if (value.StartsWith("+380") && value.Count() == 13)
                return value;
            else
                throw new ArgumentException("Phone number must be format +380XXXXXXXXX");
        }

        public static double Price(double value)
        {
            decimal d;
            if (!(decimal.TryParse(Convert.ToString(value), out d) && d >= 0 && d * 100 == Math.Floor(d * 100)))
            {
                throw new ArgumentException("Format of budget is wrong!");
            }
            return value;
        }

        public static double Price(double value1, double value2)
        {
            if (value2 <= value1)
                throw new ArgumentException("Mouthly budget must be < than yearly!");
            return value2;
        }

        public static string URL(string value)
        {
            Uri uri = null;
            if (!Uri.TryCreate(value, UriKind.Absolute, out uri) || null == uri)
                throw new ArgumentException("The website_url is not valid");
            else
                return value;
        }
    }
}
